test/test_sfinae.cpp

```diff
@@ -0,0 +1,51 @@
+/***************************************************************************
+* Copyright (c) Johan Mabille, Sylvain Corlay and Wolf Vollprecht          *
+* Copyright (c) QuantStack                                                 *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include <complex>
+#include <limits>
+
+#include "gtest/gtest.h"
+#include "xtensor/xtensor.hpp"
+#include "xtensor/xarray.hpp"
+#include "xtensor/xfixed.hpp"
+#include "xtensor/xview.hpp"
+#include "xtensor/xio.hpp"
+
+namespace xt
+{
+    template <class E>
+    inline E sfinae_rank_func(E&& a)
+    {
+        std::cout << a.rank << std::endl;
+        E b = a;
+        b.fill(0);
+        return b;
+    }
+
+    template <class E, class = typename xt::has_rank_t<E, 2>>
```

Comment:
@JohanMabille Can you help me here? ☺️

Reply:
I think the following should work:

```cpp
template <class E, std::enable_if_t<!xt::has_rank_t<E, 2>::value, int> = 0>
inline E sfinae_rank_func(E&& a)
{
        std::cout << a.rank << std::endl;
        E b = a;
        b.fill(0);
        return b;
}

```cpp
template <class E, std::enable_if_t<xt::has_rank_t<E, 2>::value, int> = 0>
inline E sfinae_rank_func(E&& a)
{
    E b = a;
    b.fill(2);
    return b;
}
```

Reply:
Thanks @JohanMabille . That's what I tried to. The thing is, I get stuck on the following:
```cpp
/Users/tdegeus/data/prog/forks/xtensor/include/xtensor/xutils.hpp:868:51: error: type 'xt::xarray_container<xt::uvector<unsigned long,
      std::__1::allocator<unsigned long> >, xt::layout_type::row_major, xt::svector<unsigned long, 4, std::__1::allocator<unsigned long>,
      true>, xt::xtensor_expression_tag> &' cannot be used prior to '::' because it has no members
        using type = std::integral_constant<bool, E::rank == N>;
```

Reply:
Ha, see my comment above.

Reply:
Ha, see my comment below.